lib/interface-types/src/interpreter/instructions/strings.rs

```diff
@@ -0,0 +1,386 @@
+use super::to_native;
+use crate::{
+    ast::InterfaceType,
+    errors::{InstructionError, InstructionErrorKind},
+    interpreter::{
+        wasm::{
+            structures::{FunctionIndex, TypedIndex},
+            values::InterfaceValue,
+        },
+        Instruction,
+    },
+};
+use std::cell::Cell;
+
+executable_instruction!(
+    string_lift_memory(instruction: Instruction) -> _ {
+        move |runtime| -> _ {
+            let inputs = runtime.stack.pop(2).ok_or_else(|| {
+                InstructionError::new(
+                    instruction,
+                    InstructionErrorKind::StackIsTooSmall { needed: 2 },
+                )
+            })?;
+
+            let memory_index: u32 = 0;
+
+            let memory = runtime
+                .wasm_instance
+                .memory(memory_index as usize)
+                .ok_or_else(|| {
+                    InstructionError::new(
+                        instruction,
+                        InstructionErrorKind::MemoryIsMissing { memory_index },
+                    )
+                })?;
+
+            let pointer = to_native::<i32>(&inputs[0], instruction)? as usize;
```

Comment:
Does the fact this is an `i32` cause a problem?  It seems that we can't access the upper 2gb of memory by casting from `i32` to `usize` because `-1i32 as usize` is `18446744073709551615`; `-1i32 as u32 as usize` is `4294967295` which seems closer to what we'd want here.

If this is a mistake, perhaps we should add some tests using the upper 2gb of memory to make sure everything is working as expected as this issue will likely be in a lot of places!

Reply:
Hmm. The pointer is of kind `i32` in the spec. It is regular a Wasm core type `i32`. Our API needs a `usize`, hence the casting.

You're right we need to check it is positive. I'm opening an issue to focus on that.